src/transports/ws.rs

```diff
@@ -134,8 +134,17 @@ impl WsServerTask {
                 let stream = async_native_tls::connect(host, stream).await?;
                 MaybeTlsStream::Tls(compat::compat(stream))
             }
-            #[cfg(not(any(feature = "ws-tls-tokio", feature = "ws-tls-async-std")))]
-            panic!("The library was compiled without TLS support. Enable ws-tls-tokio or ws-tls-async-std feature.");
+            #[cfg(all(
+                feature = "ws-rustls-tokio",
+                not(feature = "ws-tls-tokio"),
+                not(feature = "ws-tls-async-std")
```

Comment:
I think it might be a bit confusing to fallback to `ws-tls-tokio` in case multiple features are enabled. I think I'd rather explicitly panic here or ideally during compilation.

Reply:
That approach can work, but in less simple use cases it isn't always ideal. In deeper dependency trees you could have a dependency crate _A_ using _web3_ with default features, and then use it top-level with rustls.

Considering that case:
* A runtime failure seems undesirable and unexpected.
* A build error makes more sense. But can be painful to address. From the perspective of top-level you need to push _rustls_ changes upstream or fork. For the perspective of the crate _A_ maintainer changing openssl->rustls is more of a breaking change (since it can cause downstream builds to fail).
* Whereas here _web3_ continues working and just chooses _openssl_ code since both features were selected. It's easier to maintain from the outside since crate _A_ can change, e.g. to _rustls_ by default without risking breaking downstream. It is up to the user to actually ensure openssl isn't selected by any dependency, but things will generally still work.

We could also could prefer _rustls_, instead of openssl, when both are selected. I just picked what was already there for the preference.

Out of all of those Just let me know which way you'd prefer to go :slightly_smiling_face:

Reply:
@tomusdrw I'm happy to handle this case with a compile error or working (preferring one or the other), can you let me know which you want & I'll update PR.

Reply:
@alexheretic thanks for explanations and sorry for the delay. I think it's fine the way it is, your explanation makes a lot of sense.